salt/modules/win_iis.py

```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+                pscmd.append(" $Settings['{0}'] = [String] $Property;".format(setting))
+                pscmd.append(r' $Property = $Null;')
+        else:
+            availSetStr = ', '.join(availableSettings)
+
+            message = 'Unexpected setting:' + setting + '. Available settings are: ' + availSetStr
+            raise SaltInvocationError(message)
+
+    pscmd.append(' $Settings')
+    # Run commands and return data as json
+    cmd_ret = _srvmgr(cmd=str().join(pscmd), return_json=True)
+
+    # Update dict var to return data
+    try:
+        items = json.loads(cmd_ret['stdout'], strict=False)
+
+        if isinstance(items, list):
+            ret.update(items[0])
+        else:
+            ret.update(items)
+    except ValueError:
+        log.error('Unable to parse return data as Json.')
+
+    if None in ret.viewvalues():
+        message = 'Some values are empty - please validate site and web application names. Some commands are case sensitive'
+        raise SaltInvocationError(message)
+
+    return ret
+
+
+
+def set_webapp_settings(name, site, settings):
+    '''
+    Configure an IIS application.
+    .. note:
+        This function only configures existing app
+    :param str name: The IIS application.
+    :param str site: The IIS site name.
+    :param str settings: A dictionary of the setting names and their values.
+    :available settings:    physicalPath: The physical path of the webapp.
+    :                       applicationPool: The application pool for the webapp.
+    :                       userName: "connectAs" user
+    :                       password: "connectAs" password for user
+    :return: A boolean representing whether all changes succeeded.
+    :rtype: bool
+    CLI Example:
+    .. code-block:: bash
+        salt '*' win_iis.set_webapp_settings name='app0' site='site0' settings="{'physicalPath': 'C:\site0', 'apppool': 'site0'}"
+    '''
+    pscmd = list()
+    current_apps = list_apps(site)
+    current_sites= list_sites()
+    availableSettings = ('physicalPath', 'applicationPool', 'userName', 'password')
+
+    # Validate params
+    if name not in current_apps:
+        log.debug("Application %s doesn't exist", name)
+        return False
+
+    if site not in current_sites:
+        log.debug("Site %s doesn't exist", site)
+        return False
+
+    if not settings:
+        log.warning('No settings provided')
+        return False
+
+    # Treat all values as strings for the purpose of comparing them to existing values & validate settings exists in predefined settings list
+    for setting in settings.keys():
+        if setting in availableSettings:
+            settings[setting] = str(settings[setting])
+        else:
+            log.debug("Unexpected setting: %s ", setting)
+            availSetStr = ', '.join(availableSettings)
+            log.debug("Available settings: %s", availSetStr)
+            return False
+
+
+
```

Comment:
Too many blank lines here (lint error). Should have at the most one blank line within a function.